scripts/mutect2_wdl/unsupported/mutect2_opt.wdl

```diff
@@ -528,60 +683,88 @@ task MergeBamOuts {
-              wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/oncotator/oncotator_v1_ds_April052016.tar.gz
-              tar zxvf oncotator_v1_ds_April052016.tar.gz
-              ln -s oncotator_v1_ds_April052016 onco_dbdir
-          fi
-
-        ${default="/root/oncotator_venv/bin/oncotator" oncotator_exe} --db-dir onco_dbdir/ -c $HOME/tx_exact_uniprot_matches.AKT1_CRLF2_FGFR1.txt  \
-            -v ${m2_vcf} ${case_id}.maf.annotated hg19 -i VCF -o TCGAMAF --skip-no-alt --infer-onps --collapse-number-annotations --log_name oncotator.log \
-            -a Center:${default="Unknown" sequencing_center} \
-            -a source:${default="Unknown" sequence_source} \
-            -a normal_barcode:${default=" " control_id} \
-            -a tumor_barcode:${case_id} \
-            ${"--default_config " + default_config_file}
+task OncotateM2 {
+  File m2_vcf
+  String case_id
+  String? control_id
+  File? onco_ds_tar_gz
+  String? onco_ds_local_db_dir
+  String? oncotator_exe
+  String? sequencing_center
+  String? sequence_source
+  File? default_config_file
+
+  # Runtime parameters
+  Int? preemptible_attempts
+  String oncotator_docker
+  Int disk_space_gb
+  Int? mem
+
+  # Mem is in units of GB but our command and memory runtime values are in MB
+  Int machine_mem = if defined(mem) then mem * 1000 else 3500
+  Int command_mem = machine_mem - 500
+
+  command <<<
+    set -e
+
+    # local db dir is a directory and has been specified
+    if [[ -d "${onco_ds_local_db_dir}" ]]; then
+      echo "Using local db-dir: ${onco_ds_local_db_dir}"
+      echo "THIS ONLY WORKS WITHOUT DOCKER!"
+      ln -s ${onco_ds_local_db_dir} onco_dbdir
+
+    elif [[ "${onco_ds_tar_gz}" == *.tar.gz ]]; then
+      echo "Using given tar file: ${onco_ds_tar_gz}"
+      mkdir onco_dbdir
+      tar zxvf ${onco_ds_tar_gz} -C onco_dbdir --strip-components 1
+      rm ${onco_ds_tar_gz}
+
+    else
+      echo "Downloading and installing oncotator datasources from Broad FTP site..."
+      # Download and untar the db-dir
+      wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/oncotator/oncotator_v1_ds_April052016.tar.gz
+      tar zxvf oncotator_v1_ds_April052016.tar.gz
+      rm oncotator_v1_ds_April052016.tar.gz
+      ln -s oncotator_v1_ds_April052016 onco_dbdir
+    fi
+
+    ${default="/root/oncotator_venv/bin/oncotator" oncotator_exe} --db-dir onco_dbdir/ -c $HOME/tx_exact_uniprot_matches.AKT1_CRLF2_FGFR1.txt  \
+      -v ${m2_vcf} ${case_id}.maf.annotated hg19 -i VCF -o TCGAMAF --skip-no-alt --infer-onps --collapse-number-annotations --log_name oncotator.log \
+      -a Center:${default="Unknown" sequencing_center} \
+      -a source:${default="Unknown" sequence_source} \
+      -a normal_barcode:${default=" " control_id} \
+      -a tumor_barcode:${case_id} \
+      ${"--default_config " + default_config_file}
     >>>
 
     runtime {
         docker: "${oncotator_docker}"
-        memory: "3 GB"
-        bootDiskSizeGb: 12
```

Comment:
Is this not needed? Is oncotator_docker always smaller than 10G?

Reply:
It currently works and I dont think a docker should ever be larger than 10G....but if so then I guess that can looked at then?  Just irked me so I took it out